astropy/tests/runner.py

```diff
@@ -183,6 +183,15 @@ def run_tests(self, **kwargs):
 
         # Don't import pytest until it's actually needed to run the tests
         import pytest
+        if pytest.__version__ > '3.7' and pytest.__version__ < '4':
```

Comment:
actually it's incompatible with <4 as well, due to the deprecation warnings that turned into exceptions in that version.

Reply:
Also I don't think ``pytest.__version__`` is safe as once pytest reaches version 10.x this check will fail. If they go like setuptools, version 10.x is not that unlikely.

Reply:
Given how often they are releasing lately, I am not surprised if they hit 10.x in the near future.

Reply:
wouldn't LooseVersion solve this issue?